### PR TITLE
Vectorization and parallelization of PoissonLogL

### PR DESCRIPTION
**Pull request by xvallspl**

### CODE REVIEW

**Comment by vgvassilev** [comment]
Do you think we can add a gtest for `PoissonLogL`?

**Comment by xvallspl** [comment]
Sorry!

**Comment by xvallspl** [comment]
@phsft-bot build!

**Comment by xvallspl** [comment]
Rebased.

@phsft-bot  build!

**Comment by vgvassilev** [comment]
@xvallspl, could you address the clang-tidy and clang-format issues?

**Comment by xvallspl** [comment]
@vgvassilev The clang-tidy issues are addressed in my next PR

File: math/mathcore/inc/Fit/FitUtil.h [hunk 1]
```diff
@@ -478,27 +475,27 @@ namespace FitUtil {
...
-                  vecCore::Load<T>(xc[j], data.GetCoordComponent(i * vecSize, j));
-               x = xc.data();
+               std::vector<T> x(data.NDim());
+               for (unsigned int j = 0; j < data.NDim(); ++j)
+                  x[j] = *data.GetCoordComponent(i, j);
+#ifdef USE_PARAMCACHE
+                fval = func ( x.data() );
+#else
+                fval = func ( x.data(), p );
+#endif
```
Comment (amadio) [inline math/mathcore/inc/Fit/FitUtil.h]: I don't really like the `#ifdef` here and elsewhere. This is a nighmare for testing and maintaining. Isn't it possible to just define the function to always take two parameters, but ignore the second one when `USE_PARAMCACHE` is or is not defined? The logic looks backward here (looks like a parameter cache is passed when `USE_PARAMCACHE` is *not* defined).

**Pull request merged**